pywebdav/lib/WebDAVServer.py

```diff
@@ -56,6 +56,59 @@ class DAVRequestHandler(AuthServer.AuthRequestHandler, LockManager):
     server_version = "DAV/" + __version__
     encode_threshold = 1400  # common MTU
 
+    def data_to_bytes_iterator(self, data):
+        if isinstance(data, (six.string_types, six.binary_type)):
+            log.debug("Not using iterator, string type")
+            data = (data,)
+        elif self._config.DAV.getboolean('http_response_use_iterator'):
+            # Use iterator to reduce using memory
+            log.debug("Use iterator")
+        else:
+            # Don't use iterator, it's a compatibility option
+            data = (data.read(),)
+            log.debug("Don't use iterator")
+
+        for buf in data:
+            yield buf if isinstance(buf, six.binary_type) else str(buf).encode('utf-8')
```

Comment:
This would be safer as:
```suggestion
            yield buf if isinstance(buf, six.binary_type) else bytes(buf, 'utf-8') 
```
Using `str(buf)` could cause an undesired output if `buf` happens to be something that doesn't implicitely convert to the str cleanly.

Reply:
Maybe we should just throw exception in such case? This should be str or bytes anyway.

Reply:
Well I don't think it's an expected error, throwing an exception is ugly useful if you're going to catch and deal with it somewhere.

Without tracing the code through I could see it potentially being a bytearray, or a future change might want to change it to one...

This was more of a best practices suggestion, if you want to convert the object to a bytes it's better to convert in one step, rather than convert to str first and then to bytes.